plugins/feedback/src/components/EntityFeedbackPage/EntityFeedbackPage.tsx

```diff
@@ -0,0 +1,192 @@
+
+  const pluginConfig = {
+    'feedback/type': entity.metadata.annotations!['feedback/type'],
+    'feedback/host': entity.metadata.annotations!['feedback/host'],
+    'feedback/mail-to': entity.metadata.annotations!['feedback/email-to'],
+    'jira/project-key': entity.metadata.annotations!['jira/project-key'],
+  };
+
+  async function handleModalOpen() {
+    const userEntity = (await user.getBackstageIdentity()).userEntityRef;
+    setModalProps({
+      projectEntity: projectEntity,
+      userEntity: userEntity,
+      serverType: pluginConfig['feedback/type'],
+    });
+
+    return setModalOpen(true);
+  }
+
+  const handleModalClose = (respObj?: any) => {
+    setModalOpen(false);
+    if (respObj) {
+      setReload(true);
+      if (respObj.error) {
+        setSnackbarOpen({
+          message: respObj.error,
+          severity: 'error',
+          open: true,
+        });
+      } else if (respObj.message) {
+        setSnackbarOpen({
+          message: respObj.message,
+          severity: 'success',
+          open: true,
+        });
+      }
+    }
+  };
+
+  const handleSnackbarClose = (
+    event?: React.SyntheticEvent,
+    reason?: string,
+  ) => {
+    event?.preventDefault();
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackbarOpen({ ...snackbarOpen, open: false });
+  };
+
+  const handleResyncClick = () => {
+    setReload(true);
+  };
+
+  return pluginConfig['feedback/mail-to'] === undefined ? (
+    <CustomEmptyState {...pluginConfig} />
+  ) : (
+    <Grid container justifyContent="flex-end">
+      <FeedbackDetailsModal />
+      <Grid item>
+        <ButtonGroup
+          className={classes.buttonGroup}
+          color="primary"
+          variant="outlined"
+        >
+          <Tooltip
+            title="Give a feedback / Report a issue"
+            arrow
+            TransitionComponent={Zoom}
+          >
+            <Button startIcon={<Add />} onClick={handleModalOpen}>
+              Create
+            </Button>
+          </Tooltip>
+          {pluginConfig['feedback/type'] === 'JIRA' && (
+            <Tooltip
+              title="Go to Jira Project"
+              arrow
+              TransitionComponent={Zoom}
+            >
+              <Button
+                endIcon={<ArrowForwardRounded />}
+                href={`${pluginConfig['feedback/host']}/projects/${pluginConfig['jira/project-key']}/summary`}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                Go to Jira Project
+              </Button>
+            </Tooltip>
+          )}
+          <Tooltip title="Refresh" arrow TransitionComponent={Zoom}>
+            <Button onClick={handleResyncClick}>
+              {!reload ? <Sync /> : <CircularProgress size="1.5rem" />}
+            </Button>
+          </Tooltip>
+        </ButtonGroup>
+      </Grid>
+      <Dialog
+        open={modalOpen}
+        onClose={() => handleModalClose('')}
```

Comment:
maybe pass `false` to `handleModalClose` instead of an empty string here, which would then make the handler code clearer.

Reply:
I have updated this code, thanks for the review